Tests/SKCoreTests/BuildSystemManagerTests.swift

```diff
@@ -412,15 +428,18 @@ final class BuildSystemManagerTests: XCTestCase {
 // MARK: Helper Classes for Testing
 
 /// A simple `MainFilesProvider` that wraps a dictionary, for testing.
-private final class ManualMainFilesProvider: MainFilesProvider {
-  let lock: DispatchQueue = DispatchQueue(label: "\(ManualMainFilesProvider.self)-lock")
-  private var _mainFiles: [DocumentURI: Set<DocumentURI>] = [:]
-  var mainFiles: [DocumentURI: Set<DocumentURI>] {
-    get { lock.sync { _mainFiles } }
-    set { lock.sync { _mainFiles = newValue } }
+private final actor ManualMainFilesProvider: MainFilesProvider {
+  private var mainFiles: [DocumentURI: Set<DocumentURI>]
+
+  init(mainFiles: [DocumentURI: Set<DocumentURI>]) {
+    self.mainFiles = mainFiles
+  }
+
+  func setMainFiles(mainFiles: [DocumentURI: Set<DocumentURI>]) async {
+    self.mainFiles = mainFiles
   }
 
-  func mainFilesContainingFile(_ file: DocumentURI) -> Set<DocumentURI> {
+  func mainFilesContainingFile(_ file: DocumentURI) async -> Set<DocumentURI> {
```

Comment:
Does this function have to be `async`. I don’t see any `await` calls within it.

Reply:
it was copied from the protocol. let me remove it.